maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -65,7 +66,7 @@ private void resolve(StandaloneRun run) throws Exception, IOException {
 			List<Container> runBundles = projectResolver.getRunBundles();
 			Collection<Container> currentRunBundles = run.getRunbundles();
 			if (!CollectionUtils.isEqualCollection(currentRunBundles, runBundles)) {
-				if (failOnChanges) {
+				if (!resolve && failOnChanges) {
```

Comment:
This change is wrong. The resolve method is only called if the resolve field is true. So `!resolve` here always evaluates to false making `failOnChanges` meaningless.